implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/DynatraceConfig.java

```diff
@@ -64,14 +67,33 @@ default String group() {
         return get(prefix() + ".group");
     }
 
+    /**
+     * Return the version of the target Dynatrace API.
+     *
+     * @return a {@link String} containing the version of the targeted Dynatrace API.
+     */
+    default String apiVersion() {
+        // if not specified, defaults to v1 for backwards compatibility.
+        return getString(this, "apiVersion")
+                .map(val -> StringUtils.isEmpty(val) ? "v1" : val)
+                .get();
+    }
+
     @Override
     default Validated<?> validate() {
-        return checkAll(this,
-                c -> StepRegistryConfig.validate(c),
-                checkRequired("apiToken", DynatraceConfig::apiToken),
-                checkRequired("uri", DynatraceConfig::uri),
-                checkRequired("deviceId", DynatraceConfig::deviceId),
-                check("technologyType", DynatraceConfig::technologyType).andThen(Validated::nonBlank)
-        );
+        // Currently only v1 is implemented. This check will be extended once more versions are added.
+        Function<DynatraceConfig, Validated<String>> versionFunc = checkRequired("apiVersion", DynatraceConfig::apiVersion);
+        String versionStr = versionFunc.apply(this).get();
```

Comment:
You can do something like this instead so that you only change one line of this whole method:
```
checkRequired("apiVersion", DynatraceConfig::apiVersion).andThen(v -> v.invalidateWhen(<check-version-here>)),
```

Reply:
I simplified this a bit, also thanks to the suggestion of using an enum.